test/e2e_node/device_plugin.go

```diff
@@ -160,6 +162,112 @@ var _ = framework.KubeDescribe("Device Plugin [Feature:DevicePlugin] [Serial] [D
 			f.PodClient().DeleteSync(pod1.Name, &metav1.DeleteOptions{}, framework.DefaultPodDeletionTimeout)
 			f.PodClient().DeleteSync(pod2.Name, &metav1.DeleteOptions{}, framework.DefaultPodDeletionTimeout)
 		})
+		It("Verifies the Kubelet device plugin functionality with Probe-Mode plugin discovery mechanism", func() {
```

Comment:
Would it be simpler if we just remove the kubelet registration code and only keep the following part in the test?

Reply:
Looks like we do need to test both cases, but instead of having a separate similar test case, what do you think that we create a DEPRECATION file under /var/lib/kubelet/device_plugins/ directory in device manager GetWatcherCallback(), and add a helper function here that calls dp.Register only if that file doesn't exist? Perhaps even better is to do this in stub device plugin, and then we may add some test on upgrade scenario.